### DISCUSSION TITLE
Updating

### DISCUSSION THREAD

**Discussion by V6baja** [Blackymas/NSPanel_HA_Blueprint#2112] [General]
I need to update from v3.4 to the latest V.   and am look for the most up to date install instructions and any tips

Thank You

**Comment by edwardtfn** [comment]
Those are the most updated instructions: https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/howto.md#update-blueprint

**Comment by V6baja** [comment]
I though i saw that  you have to re-Flash when going to 4.3?

**Reply by edwardtfn** [reply]
Nope, but you should if you are changing framework.

Please take a look at this: https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/1756

**Comment by tomatenkopp** [comment]
Hello all,
I'm unsure about the most secure way to update ESPhome 2024.6.x / Blueprint 4.3.x
Blueprint 4.3.9 release info says: "Home Assistant v2024.6.0 and ESPHome v2024.6.0 are required" - what does this mean?
First update HA and ESP and afterwards update Blueprint to 4.3.9
OR first update blueprint so that HA and ESP Updates can be done safely?

Currently I have
HA 2024.6.3 
ESPhome 2024.5.5 
Blueprint 4.3.8

Cheers, Tom

**Reply by edwardtfn** [reply]
I usually update ESPHOME firmware first, just because that is easier to rool back, but in reality doesn't matter the order you do the update.

> Blueprint 4.3.9 release info says: "Home Assistant v2024.6.0 and ESPHome v2024.6.0 are required" - what does this mean?

That means you need the latest Home Assistant and ESPHome compiler.
